src/ethereum/frontier/eth_types.py

```diff
@@ -120,7 +120,23 @@ class Receipt:
     post_state: Root
     cumulative_gas_used: Uint
     bloom: Bloom
-    logs: List[Log]
+    logs: Tuple[Log, ...]
 
 
 State = Dict[Address, Account]
+
+
+def move_funds(
+    state: State,
+    sender_address: Address,
+    recipient_address: Address,
+    amount: U256,
+) -> None:
+    """Move funds between accounts."""
+    sender = state[sender_address]
+    assert sender.balance >= amount
+    state[sender_address] = replace(sender, balance=sender.balance - amount)
+    recipient = state[recipient_address]
+    state[recipient_address] = replace(
+        recipient, balance=recipient.balance + amount
+    )
```

Comment:
I personally prefer my suggestion from earlier which was:

```python
state.update(address, balance=lambda x: x + BLOCK_REWARD, nonce=lambda x: x + 1)
```

vs your

```python
def make_changes(account: MutableAccount):
    account.nonce += 1
    account.balance += BLOCK_REWARD

state.update(address, make_changes)
```

Your version is more work to implement because it requires rewriting `dataclasses.dataclass` to produce a immutable and mutable version of each class.

Reply:
I find @SamWilsn's approach more readable.